src/components/pages/import.js

```diff
@@ -0,0 +1,158 @@
+
+function Import({ initialState, importApiClient }) {
+  const { t } = useTranslation('import');
+
+  const { language } = useLanguage();
+  const { formatDate } = useDateFormat();
+  const languageNameProvider = useService(LanguageNameProvider);
+
+  const sourceMenuItems = initialState.importSources;
+  const [selectedSource, setSelectedSource] = useState(null);
+  const [selectedDocumentKeys, setSelectedDocumentKeys] = useState([]);
+  const [importableDocuments, setImportableDocuments] = useState([]);
+
+  const handleImportClick = () => {
+    const tasks = selectedDocumentKeys
+      .map(key => importableDocuments.find(doc => doc.key === key))
+      .map(doc => ({
+        key: doc.key,
+        importedRevision: doc.importedRevision,
+        importableRevision: doc.importableRevision
+      }));
+    logger.info(`Dummy import sets tasks: ${JSON.stringify(tasks)}`);
+  };
+
+  const handleSourceMenuChange = async value => {
+    const newSelectedSource = sourceMenuItems.find(source => source.name === value);
+    setSelectedSource(newSelectedSource);
+    setImportableDocuments([]);
+    setSelectedDocumentKeys([]);
+
+    const { documents } = await importApiClient.getImports(newSelectedSource);
+    setImportableDocuments(documents);
+  };
+
+  const renderImportType = doc => {
+    let icon = doc.importType;
+    if (doc.importType === DOCUMENT_IMPORT_TYPE.add) {
+      icon = <CloudDownloadOutlined />;
+    }
+    if (doc.importType === DOCUMENT_IMPORT_TYPE.update) {
+      icon = <CloudSyncOutlined />;
+    }
+    return <Tooltip title={t(doc.importType)} className="ImportPage-importType">{icon}</Tooltip>;
+  };
+
+  const renderTitle = doc => {
+    if (!doc.slug) {
+      return <span>{doc.title}</span>;
+    }
+
+    const url = `${selectedSource.baseUrl}${urls.getArticleUrl(doc.slug)}`;
+    return <a href={url}>{doc.title}</a>;
+  };
+
+  const renderLanguage = doc => {
+    const langaugesData = languageNameProvider.getData(language);
+    const documentLanguageData = langaugesData[doc.language];
+    return <CountryFlagAndName code={documentLanguageData.flag} name={documentLanguageData.name} flagOnly />;
+  };
+
+  const renderUpdateDate = doc => {
+    return <span>{formatDate(doc.updatedOn)}</span>;
+  };
+
+  const columns = [
+    { title: t('importType'), key: 'importType', width: '150px', sorter: by(x => x.importType), render: renderImportType },
+    { title: t('title'), key: 'title', sorter: by(x => x.title), render: renderTitle },
+    { title: t('language'), key: 'language', width: '150px', sorter: by(x => x.language), render: renderLanguage },
+    { title: t('updateDate'), key: 'updateDate', width: '200px', sorter: by(x => x.updatedOn), render: renderUpdateDate }
+  ];
+
+  const tableRowSelection = {
+    onChange: selectedRowKeys => {
+      setSelectedDocumentKeys(selectedRowKeys);
+    }
+  };
+
+  const showSpinner = !!selectedSource && importableDocuments.length === 0;
```

Comment:
This looks like the spinner will load indefinitely when there is nothing to import on the source system, thus, looking like there's still something going on to the user. We might need an extra `isFetchingImportableDocuments` or similar as a state in this component.

Reply:
Of course, true, did not realise while playing with mock data so much.
Pretty nice this `no-data` attribute, did not know about it and couldn't find it in the ant.design documentation.
Changed as suggested, it's also nicer and simpler now 👍